server/sonar-server/src/main/java/org/sonar/server/settings/ws/SetAction.java

```diff
@@ -203,6 +301,24 @@ private static SetRequest toWsRequest(Request request) {
     return setRequest;
   }
 
+  private static List<Map<String, String>> readFieldValues(Request request) {
+    String key = request.mandatoryParam(PARAM_KEY);
+
+    return request.multiParam(PARAM_FIELD_VALUES).stream()
+      .map(json -> readOneFieldValues(json, key))
+      .collect(Collectors.toList());
+  }
+
+  private static Map<String, String> readOneFieldValues(String json, String key) {
+    Type type = new TypeToken<Map<String, String>>() {
+    }.getType();
+    try {
+      return (Map<String, String>) GsonHelper.create().fromJson(json, type);
+    } catch (JsonParseException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)